pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -3393,13 +3395,14 @@ func handleIBDelete(ctx *domainContext, name string) {
 			ib.IsPCIBack = false
 		}
 	}
+	// Create a new list with everything but "ib" included
 	replace := types.AssignableAdapters{Initialized: true,
 		IoBundleList: make([]types.IoBundle, len(aa.IoBundleList)-1)}
 	for _, e := range aa.IoBundleList {
 		if e.Type == ib.Type && e.Name == ib.Name {
 			continue
 		}
-		replace.AddOrUpdateIoBundle(e)
+		replace.IoBundleList = append(replace.IoBundleList, e)
```

Comment:
Why not use AddOrUpdate? I understand there is no functional impact.. But probably always use that routine?

Reply:
Because it is nonsensical to use AddOrUpdate when the whole thing is constructed from scratch. And it raised the question of which fields should be preserved in that case.